### ISSUE TITLE
DCGM exporter crashloopbackoff

### ISSUE THREAD

**Issue by faust64** [NVIDIA/dcgm-exporter#37]
Hi,

Using the nvidia GPU operator, on OpenShift (4.7.37).
As of a couple weeks ago, when deploying new GPU nodes, the dcgm-exporter DaemonSet creates Pods that would crash on a loop.
We have several clusters affected. Running the same OpenShift version, same version of the GPU operator (v1.8.2), same ClusterPolicy object.

Failing Pods would pull their image from `nvcr.io/nvidia/k8s/dcgm-exporter:2.2.9-2.5.0-ubi8`
Works on nodes that resolved this to `nvcr.io/nvidia/k8s/dcgm-exporter@sha256:fd1c82078e67368b49baa69e4c0644325fb2bda8ca98833cd1794f2b7e8f7f16`
Broken on nodes running that version:  `nvcr.io/nvidia/k8s/dcgm-exporter@sha256:9264ed31a3190de36ce652c3b8d35ee3b0b14bcac50b4451b58c809e93004785`

Pods in crashloopbackoff shows the following logs:
```
$> oc logs -n gpu-operator-resources nvidia-dcgm-exporter-4s2g9 -p
time="2021-12-22T14:52:20Z" level=info msg="Starting dcgm-exporter"
time="2021-12-22T14:52:20Z" level=info msg="Attemping to connect to remote hostengine at 100.79.3.83:5555"
time="2021-12-22T14:52:20Z" level=fatal msg="Error connecting to nv-hostengine: API version mismatch"
```

All other pods in that namespace are OK, ready/alive.

ClusterPolicy:

```
spec:
  daemonsets:
    priorityClassName: xxx
    tolerations:
    - operator: Exists
  dcgm:
    image: dcgm
    repository: nvcr.io/nvidia/cloud-native
    version: 2.2.9-ubi8
  dcgmExporter:
    image: dcgm-exporter
    repository: nvcr.io/nvidia/k8s
    version: 2.2.9-2.5.0-ubi8
  devicePlugin:
    image: k8s-device-plugin
    repository: nvcr.io/nvidia
    version: v0.9.0-ubi8
  driver:
    image: driver
    manager:
      env:
      - name: ENABLE_AUTO_DRAIN
        value: "true"
      - name: DRAIN_USE_FORCE
        value: "false"
      - name: DRAIN_POD_SELECTOR_LABEL
        value: ""
      - name: DRAIN_TIMEOUT_SECONDS
        value: 0s
      - name: DRAIN_DELETE_EMPTYDIR_DATA
        value: "false"
      image: k8s-driver-manager
      repository: nvcr.io/nvidia/cloud-native
      version: v0.1.0
    repository: nvcr.io/nvidia
    version: 470.57.02
  gfd:
    image: gpu-feature-discovery
    repository: nvcr.io/nvidia
    version: v0.4.1
  migManager:
    enabled: false
    image: k8s-mig-manager
    repository: nvcr.io/nvidia/cloud-native
    version: v0.1.3-ubi8
  nodeStatusExporter:
    image: gpu-operator-validator
    repository: nvcr.io/nvidia/cloud-native
    version: v1.8.2-ubi8
  operator:
    defaultRuntime: crio
    initContainer:
      image: cuda
      repository: nvcr.io/nvidia
      version: 11.3.0-base-ubi8
    runtimeClass: nvidia
  toolkit:
    image: container-toolkit
    repository: nvcr.io/nvidia/k8s
    version: 1.7.1-ubi8
  validator:
    image: gpu-operator-validator
    repository: nvcr.io/nvidia/cloud-native
    version: v1.8.2-ubi8
```

Inspecting the image tags, I don't see anything unexpected, like a mismatching version ... Not sure what would be going on.

Besides, that CRD schema won't allow me to define some sha256 formatted version for my dgcm-exporter image ...
What can I do?
Is this a known issue pending some fix?

**Comment by faust64** [comment]
... well... not sure why, editing that DaemonSet directly, forcing the image to nvcr.io/nvidia/k8s/dcgm-exporter@sha256:fd1c82078e67368b49baa69e4c0644325fb2bda8ca98833cd1794f2b7e8f7f16, I don't see the gpu operator trying to revert my change. Which is another bug, IMHO .... But in 6 months dealing with that operator ... I stopped counting ...

The reconciles finally shows my dcgm-exporter daemonset is healthy.
Any chance you could test your images, before publishing them over another that's been working fine for months?
Lucky me, you didn't prune the previous one. yet ....

**Comment by glowkey** [comment]
Apologies for the hassle. While 2.3.x images have a fix for the "API version mismatch" problem we have also published 2.2.9-2.5.1 with a fix for this issue. A significant effort will be made to not overwrite existing releases.
